### PR TITLE
Map Updates: abandoned barn (phase 1)

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds new furnitures, nested mapgen, palettes, and more content to abandoned barns"```

#### Purpose of change
Mapgen has come a long way since the first pass into JSON and there's a lot of map updating to do. I'm going to go through the list in alphabetical order, adding the following features:
- Mapgen palettes, where appropriate
- Nested mapgen to allow variations of different buildings
- Chances of NPC survivor holdouts, or specialized zombie spawns
- Much higher loot spawn rates in most buildings, to bring them up to realistic levels
- Integrate new furniture and terrain types that have been added since the maps were made

#### Describe the solution
Abandoned Barn is alphabetically the first map and also really old.
I want to update its mapgen to use palettes, chunks, and new furnitures.
Then I want to add some unique variations, like camped out survivor NPCs and such.

I'd also like to add z-levels to some of the barns. This is a trickier proposition that would require each barn getting its own separate overmap entry, and a change to how they spawn. Still, not a bad idea... it would also allow me to make some of them span a few OM tiles and include other rundown sheds and things.

EDIT: couldn't get to it in this PR without bloating it too far.

#### Additional context
Progress list:
- [x] Convert old file to modern format
- [x] Add current furniture and terrain entries
- [x] Add some vehicle/animal/monster spawns
- [x] Update/add itemgroups
- [x] Add nested mapgen support including some NPC holdouts
- [x] Add z-levels - cancelled for this one, it's getting too big.

#### Next PR:
Add z-levels, and add unique dialogue for holdout NPCs

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
> Much higher loot spawn rates in most buildings, to bring them up to realistic levels

In my opinion this should wait until we implement some mechanism to gradually reduce items spawn rate over time (to simulate marauders and other survivors looting places). Otherwise we'll have buildings full of loot even after several years.

**Comment by I-am-Erk** [comment]
I don't think automated looting is too far down the road; Kevin is working on ai now, and was the one who suggested it. Besides, artificial scarcity of items that shouldn't be scarce is a poor game balance to rely on, and we need to see how things shake out with that removed.

**Comment by Night-Pryanik** [comment]
> artificial scarcity of items that shouldn't be scarce

I'm talking not about rocks in the field, but about food, drugs and the like instead.

**Comment by I-am-Erk** [comment]
As am I. Food shouldn't be scarce. The world ended five days ago. While the supermarkets might have been looted, nobody had any time to eat all that food or drink all that water. It didn't just vanish.

**Comment by Night-Pryanik** [comment]
But the game can be started on the next year, and it would be strange to find supermarkets unlooted at that time.

**Comment by I-am-Erk** [comment]
Sure? But that's not the standard situation, and I think the game should be balanced to the default start and then systems added in to work out niche issues, not balanced for niche issues. The default start is 5 days after the Cataclysm, and currently the spawn rate looks like the world was thoroughly looted.

**Comment by Night-Pryanik** [comment]
The major problem is not in too high or too low loot spawn rate per se, but that amount of loot remains the same no matter how much time passed after the game start.
Loot spawn rate might be increased, simulating only 5 days passed after the Cataclysm, but loot should be more and more rare as time goes by.

But we're talking about problems not related directly to the PR, there should be a separate issue on this matter.

**Comment by I-am-Erk** [comment]
Yeah, this really has nothing to do with abandoned barns - nor with my overall project to bring loot spawns to a level that makes it look like the city was slightly looted but the looters died before using it. We definitely need a system to change loot spawns over time, ideally by having npcs do the looting.

**Comment by SirPendrak** [comment]
I am very strongly aganist loot ammount decreasing over time. Unless there are super A.I. NPC survivors that are as effective as players, it sounds super cheesy that stores, houses etc. in the middle of zombie infested towns are getting emptied without killing all the zeds that guard all the loot. For places far from dangerous zones it could make sense, but probably would not add any value to the game, only tedium.

**Comment by I-am-Erk** [comment]
The proposed mechanism for loot decreasing over time is an NPC based loot calculator, where they actually do go out and get loot from the areas with fewer zombies in them.

Can we please actually talk about my pull request now? I can't think what this conversation possibly has to do with what I'm doing with this abandoned barn.

File: data/json/mapgen/nested/rural_outdoors_nested.json [hunk 1]
```diff
@@ -0,0 +1,398 @@
+[
```
Comment (ZhilkinSerg) [inline data/json/mapgen/nested/rural_outdoors_nested.json]: This file should be [linted](http://dev.narc.ro/cataclysm/format.html).

File: data/json/npcs/holdouts/rural.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[
+  {
+    "type": "npc",
+    "id": "rural_holdout_solo",
+    "//": "A rural holdout survivor living in a barn or a forest or something, all by themselves.  At some point this NPC should get special dialogue.",
+    "class": "NC_COWBOY",
+    "attitude": 0,
+    "name_suffix": "Survivor"
```
Comment (ZhilkinSerg) [inline data/json/npcs/holdouts/rural.json]:
```suggestion
    "name_suffix": "Survivor",
```

**Pull request merged**